plugins/scaffolder/src/components/fields/EntityPicker/EntityPicker.tsx

```diff
@@ -123,20 +123,28 @@ export const EntityPicker = (props: EntityPickerProps) => {
         }
       }
     },
-    [onChange, formData, defaultKind, defaultNamespace, allowArbitraryValues],
+    [
+      onChange,
+      formData,
+      defaultKind,
+      defaultNamespace,
+      allowArbitraryValues,
+      entityPresentationApi,
+    ],
   );
 
   // Since free solo can be enabled, attempt to parse as a full entity ref first, then fall
   // back to the given value.
   const selectedEntity =
-    entities?.find(e => stringifyEntityRef(e) === formData) ??
-    (allowArbitraryValues && formData ? getLabel(formData) : '');
+    entities?.find(
+      e => entityPresentationApi.forEntity(e).snapshot.entityRef === formData,
```

Comment:
You probably don't want to call this over and over; just return everything you need out of the `useAsync` above instead so that it's statically available for the renderer here. Maybe change its return type to `Array<{ entity: Entity, label: string }>` or something like that, depends on how you need it to be called

Reply:
In this case wouldn't it be better to keep `stringifyEntityRef` ? I removed it because I thought it was supposed to be replaced by presentation api but in your [previous comment](https://github.com/backstage/backstage/pull/24081#discussion_r1557383061) you suggested to use it 🙂 

It seems to be working fine with `stringifyEntityRef` 🙂